src/pybind/mgr/dashboard_v2/frontend/src/app/shared/datatable/table-key-value/table-key-value.component.ts

```diff
@@ -49,15 +39,58 @@ export class TableKeyValueComponent implements OnInit {
   ngOnInit() {
     this.columns = [
       {
-        prop: this.key,
+        prop: 'key',
         flexGrow: 1,
         cellTransformation: CellTemplate.bold
       },
       {
-        prop: this.value,
+        prop: 'value',
         flexGrow: 3
       }
     ];
+    this.useData();
+  }
+
+  ngOnChanges(changes) {
+    this.useData();
+  }
+
+  useData() {
+    let temp = [];
+    if (!this.data) {
+      return; // Wait for data
+    } else if (_.isArray(this.data)) {
+      const first = this.data[0];
+      if (_.isObject(first) && _.has(first, 'key') && _.has(first, 'value')) {
```

Comment:
we should replace `_.isObject` with `_.isPlainObject`.
`_.isObject` will return true if you test an array.

Reply:
It should be false if it is an array ;) See line 67 :)